keras/engine/training.py

```diff
@@ -1458,7 +1458,10 @@ def _standardize_user_data(self, x, y,
         output_shapes = []
         for output_shape, loss_fn in zip(self._feed_output_shapes, self._feed_loss_fns):
             if loss_fn is losses.sparse_categorical_crossentropy:
-                output_shapes.append(output_shape[:-1] + (1,))
+                if K.image_data_format() == 'channels_last':
+                    output_shapes.append(output_shape[:-1] + (1,))
+                else:
```

Comment:
Please only do this when `len(output_shape) in {4, 5}`. It *could* be that we have a 3D (timeseries) output while the user has set `image_data_format = "channels_first"`. The class axis is still -1 in this case.